website/blog/2020-12-15-certificate-pinning.md

```diff
@@ -0,0 +1,58 @@
+---
+title: "Certificate Pinning Case Study"
+author: "Radhakrishnan Asokan"
+---
+
+A case study on certificate pinning.
+
+<!--truncate-->
+
+### Introduction
+
+In HTTPS handshake, the server responds with a public certificate issued by Certificate Authority (CA) to establish a connection. If CA or Root certificate was compromised and issues a certificate without the domain owner's consent, the client may face the man-in-the-middle (MTM) attack. To protect users from this vulnerability, an app can employ SSL pinning.
+
+![MITM](/learn/assets/blog_certificate_pinning/mitmdiagram.png)
+
+Certificate Pinning is an additional layer of security which protects communication between client and server. The Standard HTTPS verifies whether the connection is secure but it cannot verify whether you are communicating with the actual server or an intercepted server.
+
+## What Certificate Pinning offers
+
+Standard HTTPS checks only whether the connection is secure. Along with the HTTPS, we use certificate pinning which will verify whether you are communicating with the actual server.
+
+The application contains the pinned certificates which are pre-defined "known" certificates. During the time of communication between the client and server, you expect that the server certificate should match with any one of the pinned certificates. If it does not match, we terminate the communication.
+
+![Certificate Pinning](/learn/assets/blog_certificate_pinning/nomitmdiagram.png)
+
+### Web Certificate Pinning
+
+Web Certificate Pinning is dynamic pinning. Certificates are pinned during the initial connection establishment. It was [introduced](https://developer.mozilla.org/en-US/docs/Web/HTTP/Public_Key_Pinning) in 2015 soon after it was deprecated in 2018 due to which it created a new set of vulnerabilities than resolving the actual [problem](https://scotthelme.co.uk/using-security-features-to-do-bad-things/).
+
+### Mobile Certificate Pinning
+
+Mobile certificate pinning is static pinning, in which the certificate bundles with the app during the release and expects to deploy a new release on the certificate expiry.
+
+## Advantages of Certificate Pinning
+
+* Certificate Pinning protects data tampering even if the user installs a malicious CA with or without knowing.
+* If a trusted certificate authority gets compromised due to security [vulnerability](https://en.wikipedia.org/wiki/Certificate_authority#CA_compromise), the application will not get affected.
+
+## Problem Associated with Certificate Pinning
+
+A security researcher has [warned](https://scotthelme.co.uk/im-giving-up-on-hpkp/) the risk associated with using HTTP Certificate Pinning.
+
+* If the key was accidentally deleted, stolen, hacked, you may face serious application [downtime issues](https://www.smashingmagazine.com/be-afraid-of-public-key-pinning/).
+* Domain hijacking - on Domain-hijacking, you can lose control of certificates, and hijackers can mishandle your certificates.
+
+## What next?
```

Comment:
I think we should add that very few sites actually use certificate pinning and one needs to be very careful in evaluating using this feature. After careful consideration, at this point, we have not yet implemented support for certificate pinning in applications build using WaveMaker.

Reply:
Added the suggested lines under "Problem Associated with Certificate Pinning", Please review again.